### PR TITLE
Address JOSS review comments

### PR DESCRIPTION
**Pull request by Yurlungur**
## PR Summary

In the the JOSS review #391 @parikshitbajpai points out two issues. The first is that the examples directory is not documented. The second is that the paper seems to blend the statement of need and state of the field sections. I address both comments here:

- For the exampled directory, I add a page to the documentation explaining how to build and run the examples, as well as what they do. I also noticed a few problems, which I resolve:
  - The `customizing` page of the docs was not properly indexed and was not showing up in HTML builds.
  - The `CMake` for the examples folder was actually commented out in the build system. This must have been an intermediate debugging change that somehow made it into main.
  - The example that gets sound speed and pressure must not have been built in a while because it showed usage of the deprecated `EOSBuilder` machinery, which no longer exists.
- For the paper, I merge the two sections and simply call the joint section `statement of need and state of the field`.

## PR Checklist

- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Format your changes by using the `make format` command after configuring with `cmake`.
- [ ] Document any new features, update documentation for changes made.
- [ ] Make sure the copyright notice on any files you modified is up to date.
- [ ] After creating a pull request, note it in the CHANGELOG.md file.
- [ ] LANL employees: make sure tests pass both on the github CI and on the Darwin CI

If preparing for a new release, in addition please check the following:
- [ ] Update the version in cmake.
- [ ] Move the changes in the CHANGELOG.md file under a new header for the new release, and reset the categories.

### CODE REVIEW

File: doc/sphinx/src/building.rst [hunk 1]
```diff
@@ -108,6 +108,7 @@ The main CMake options to configure building are in the following table:
  ``SINGULARITY_BUILD_CLOSURE``           OFF      Build the mixed cell closure models
  ``SINGULARITY_BUILD_TESTS``             OFF      Build test infrastructure.
  ``SINGULARITY_BUILD_PYTHON``            OFF      Build Python bindings.
+ ``SINGULARITY_BUILD_PYTHON``            OFF      Build examples of ``singularity-eos`` in use.
```
Comment (jhp-lanl) [inline doc/sphinx/src/building.rst:111]: `SINGULARITY_BUILD_EXAMPLES` not `SINGULARITY_BUILD_PYTHON`

File: doc/sphinx/src/examples.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  // Loop through the cells and use the two function calls
+  for (int i = 0; i < Ncells; ++i) {
+    double sie = robust::ratio(uu[i], rho[i]); // convert to specific internal energy
+    P[i] = eos.PressureFromDensityInternalEnergy(rho[i], sie, lambda.data());
+    double bmod = eos.BulkModulusFromDensityInternalEnergy(rho[i], sie, lambda.data());
+    cs[i] = std::sqrt(robust::ratio(bmod, rho[i]));
+  }
+
+The exact same code is implemented via the python bindings in ``get_sound_speed_press.py``.
```
Comment (jhp-lanl) [inline doc/sphinx/src/examples.rst:40]: It's not a python file, right?

**Review by jhp-lanl — APPROVED** [review]
Minor typos.

I skimmed the actual HTML documentation build and it looks good, but please give it a check of your own to ensure it looks as you intend it to look
https://lanl.github.io/singularity-eos/jmm/document-examples/src/examples.html#examples

File: doc/sphinx/src/examples.rst (hunk 2 above)
Comment (jhp-lanl) [reply doc/sphinx/src/examples.rst:40]: Never mind! I missed the python bindings part of this sentence.

File: doc/sphinx/src/examples.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
+.. _examples:
+
+Examples
+=========
+
+The ``examples`` directory of ``singularity-eos`` contains several
```
Comment (parikshitbajpai) [inline doc/sphinx/src/examples.rst:6]:
```suggestion
The ``example`` directory of ``singularity-eos`` contains several
```

The directory is `example` so just a minor change for consistency or if you want to rename it to examples that works as well.

**Review by parikshitbajpai — COMMENTED** [review]
This looks great to me and addresses both my comments in #391. A very minor suggestion to fix the directory name but otherwise it looks great.

File: doc/sphinx/src/examples.rst (hunk 3 above)
Comment (Yurlungur) [reply doc/sphinx/src/examples.rst]: good catch. Thanks!

File: doc/sphinx/src/building.rst (hunk 1 above)
Comment (Yurlungur) [reply doc/sphinx/src/building.rst:111]: Oops! Good catch!

**Comment by Yurlungur** [comment]
> Minor typos.
> 
> I skimmed the actual HTML documentation build and it looks good, but please give it a check of your own to ensure it looks as you intend it to look https://lanl.github.io/singularity-eos/jmm/document-examples/src/examples.html#examples

Yep. I actually renewed my practice of building the html locally. This helped me find a bunch of problems I fixed here. :)

**Comment by junghans** [comment]
That looked good to me as well.

**Pull request merged**